File: Matérias/AEDS 2/BeeCrowd/ArrayHash.c
#include <stdio.h>
#include <string.h>


int main() {
    int N;
    scanf("%i", &N);
    int L;
    char palavra[255];
    for(int i = 0; i < N; i++)
    {
        int soma = 0;
        scanf("%i", &L);
        for(int j = 0; j < L; j++)
        {
            scanf(" %[^\n]", palavra);
            for(int k = 0; k < strlen(palavra); k++)
            {
                soma += (palavra[k] - 'A' + j + k);
            }
        }
        printf("%i\n", soma);
    }


    return 0;
}

File: Matérias/AEDS 2/BeeCrowd/ArrayHash.java

import java.util.Scanner;

public class ArrayHash {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int L;
        String palavra;
        for(int i = 0; i < N; i++)
        {
            int soma = 0;
            L = scanner.nextInt();
            scanner.nextLine(); // Consume the leftover newline
            for(int j = 0; j < L; j++)
            {
                palavra = scanner.nextLine();
                    for(int k = 0; k < palavra.length(); k++) {
                        soma += (palavra.charAt(k) - 'A' + j + k);
                    }
                }
                System.out.println(soma);
            }
            scanner.close();
        }
    }

